chimney/src/main/scala/io/scalaland/chimney/internal/macros/dsl/TransformerFDefinitionWhiteboxMacros.scala

```diff
@@ -151,9 +151,31 @@ class TransformerFDefinitionWhiteboxMacros(val c: whitebox.Context) extends Macr
   ](f: Tree): Tree = {
     val To = weakTypeOf[To]
     val Inst = weakTypeOf[Inst]
+    val (instType, instSymbol) = if (Inst.typeSymbol.isJavaEnum) {
```

Comment:
This piece of code, and the one below it almost fully repeat the macro defined in `TransformerDefinitionWhiteboxMacros`
https://github.com/scalalandio/chimney/pull/206/files#diff-49b1d3d695f343c7b05d0714a4eb297fc1dd125d30150f4eec1fcf1b87208af5R105

I just copied the code to follow the former structure (every method in the API has a corresponding macro). What do you think about re-factoring to extract the common pieces and thus avoid  the duplication and the need for exhaustive testing?